### PR TITLE
Improve filter support

### PR DESCRIPTION
**Pull request by maxhille**
The changes in this PR improve filter handling.

* Support Date columns / filters
* Fix / extend onFilterChanged type decoding/encoding
* Add new example page where these improvements are demonstrated

### CODE REVIEW

File: src/AgGrid/ValueFormat.elm [hunk 1]
```diff
@@ -167,6 +168,25 @@ numberFilterValueGetter field =
...
+    dateFilterValueGetter "dateField"
+    > new Date("2023-12-12")
+
+-}
+dateFilterValueGetter : String -> String
+dateFilterValueGetter field =
+    String.Interpolate.interpolate """
+        if (!data) { return null; }
+
+        return new Date(data.birthday)
```
Comment (mschindlerMM) [inline src/AgGrid/ValueFormat.elm:186]: The field needs to be substituted.

Comment (mschindlerMM) [inline src/AgGrid/ValueFormat.elm:186]: The example didn't work for me when I tried checking for equality. But greatherThan/lessThan worked fine. I suspect that the time might be a bit off due to timezones.

**Review by mschindlerMM — CHANGES_REQUESTED** [review]

File: src/AgGrid/ValueFormat.elm (hunk 1 above)
Comment (maxhille) [reply src/AgGrid/ValueFormat.elm]: I expected AgGrid calls `new Date()` just as our ValueFormatter does (thus both being the same time zone), but I did not expect this:

screenshot-2023-12-13T14:07:54+01:00

Comment (maxhille) [reply src/AgGrid/ValueFormat.elm]: Missed that indeed. Good catch!

File: src/AgGrid/ValueFormat.elm [hunk 2]
```diff
@@ -167,6 +168,25 @@ numberFilterValueGetter field =
...
+    dateFilterValueGetter "dateField"
+    > new Date("2023-12-12T00:00")
+
+-}
+dateFilterValueGetter : String -> String
+dateFilterValueGetter field =
+    String.Interpolate.interpolate """
+        if (!data.{0}) { return null; }
+
+        return new Date(data.{0} + "T00:00")
```
Comment (mschindlerMM) [inline src/AgGrid/ValueFormat.elm:186]: Now my only fear is that we might create incorrect DateTime strings if the user provides more than just the date part.

I would say that it is not uncommon for `Date` values to be represented with a POSIX on the frontend and when converting the POSIX to an ISO8601 string for the renderer we would include the time part. This potential for error is not really obvious to the user.

Maybe we can use `setHour` instead (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setHours). This allows us to pass whatever the users provides to the `new Date` call, create a Date from it, and forcibly shift the time to midnight.

**Review by mschindlerMM — CHANGES_REQUESTED** [review]

**Review by mschindlerMM — APPROVED** [review]

**Pull request merged**